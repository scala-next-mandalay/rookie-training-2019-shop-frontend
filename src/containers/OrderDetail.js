import { connect } from 'react-redux';
import OrderDetail from '../components/OrderDetail';


const _getOrderById = (orders, id) => {

  for (const order of orders) {
    if (order.id === id) {
      return order;
    }
   }
   return null;
};
export default connect(
  
  (state) => ({
    
    orderitems: state.orderitems.rows,
    selectedOrder:  _getOrderById(state.orders.rows, state.orders.selectedOrderId)
   
  })
  
)(OrderDetail);
