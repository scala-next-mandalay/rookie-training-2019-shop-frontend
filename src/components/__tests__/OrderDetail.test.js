import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import { act } from 'react-dom/test-utils';
import OrderDetail from '../OrderDetail';
/*global expect*/
/*global jest*/
/*global Event*/

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});
const orderitems = [{ id: 1,order_id:5,item_id:12, name: "item1",quantity:2,unit_price:800,total_price:1600,},
                    { id: 2,order_id:5,item_id:10, name: "item2" ,quantity:5,unit_price:500,total_price:2500, }];

const selectedOrder ={
  first_name: "khin",
  last_name: "lawoon",
  address1: "23street",
  address2: "CMTZ",
  country: "myanmar",
  state: "mandalay",
  city: "mandalay",
  created_at :"2019-07-30 07:48:47",
  total_price: 4100
}
describe("OrderDetail component", () => {
  it('matches the snapshot', () => {
    const OrderDetailSnapshotWithSelectedOrder = renderer.create(
      <Parent><OrderDetail 
      orderitems={orderitems}
      selectedOrder={selectedOrder}
     /> </Parent>
    ).toJSON();
    expect(OrderDetailSnapshotWithSelectedOrder).toMatchSnapshot();
  });
  
  
    it('matches the snapshot', () => {
    const OrderDetailSnapshot = renderer.create(
      <Parent><OrderDetail 
      orderitems={orderitems}
      selectedOrder={null}
      /> </Parent>
    ).toJSON();
    expect(OrderDetailSnapshot).toMatchSnapshot();

});
  
});


describe("Testing OrderDetail  Component", () => {
 it('testing and checking recieved data of  order detail page', () => {
    act(() => {
      ReactDOM.render((<Parent><OrderDetail
      selectedOrder={selectedOrder}
      orderitems={orderitems}
      
     /> 
     </Parent>), container);
   
    });
    
    //table head
    const thArr = container.querySelectorAll('table th');
    expect(thArr[0].textContent.toUpperCase()).toBe('ID');
    expect(thArr[1].textContent.toUpperCase()).toBe('ORDERID');
    expect(thArr[2].textContent.toUpperCase()).toBe('ITEM ID');
    expect(thArr[3].textContent.toUpperCase()).toBe('ITEM NAME');
    expect(thArr[4].textContent.toUpperCase()).toBe('PRICE');
    expect(thArr[5].textContent.toUpperCase()).toBe('QTY');
    expect(thArr[6].textContent.toUpperCase()).toBe('TOTAL PRICE');
    
    const tdArr = container.querySelectorAll('table td');
    //first row-----
    expect(tdArr[0].textContent).toBe("1");
    // expect(tdArr[1].textContent).toBe("5");
    expect(tdArr[2].textContent).toBe("12");
    expect(tdArr[3].textContent).toBe("item1");
    expect(tdArr[4].textContent).toBe("800");
    expect(tdArr[5].textContent).toBe("2");
    expect(tdArr[6].textContent).toBe("1600");
    
    //second row----
    expect(tdArr[7].textContent).toBe("2");
    // expect(tdArr[1].textContent).toBe("5");
    expect(tdArr[9].textContent).toBe("10");
    expect(tdArr[10].textContent).toBe("item2");
    expect(tdArr[11].textContent).toBe("500");
    expect(tdArr[12].textContent).toBe("5");
    expect(tdArr[13].textContent).toBe("2500");
    
    
    const divArr2 = container.querySelectorAll('.text2 div');
    expect(divArr2[0].textContent).toBe("Order Date- 2019-07-30 07:48:47");
    expect(divArr2[1].textContent).toBe("Order Total- 4100");
    expect(divArr2[2].textContent.toUpperCase()).toBe('BILLING ADDRESS');
    expect(divArr2[3].textContent).toBe("khin lawoon");
    expect(divArr2[4].textContent).toBe("23street CMTZ");
    expect(divArr2[5].textContent).toBe("mandalay");
    expect(divArr2[6].textContent).toBe("mandalay");
    expect(divArr2[7].textContent).toBe("myanmar");
    expect(divArr2[9].textContent.toUpperCase()).toBe('PAYMENT METHOD');
    expect(divArr2[10].textContent.toUpperCase()).toBe('CHECKMONEY/MONEYORDER');
    expect(divArr2[11].textContent.toUpperCase()).toBe('DELIVERY ADDRESS');
    expect(divArr2[12].textContent).toBe("khin lawoon");
    expect(divArr2[13].textContent).toBe("23street CMTZ");
    expect(divArr2[14].textContent).toBe("mandalay");
    expect(divArr2[15].textContent).toBe("mandalay");
    expect(divArr2[16].textContent).toBe("myanmar");
    expect(divArr2[18].textContent.toUpperCase()).toBe('SHIPPING METHOD');
    expect(divArr2[19].textContent.toUpperCase()).toBe('SHIPPING');
   
    
 });
    
});
