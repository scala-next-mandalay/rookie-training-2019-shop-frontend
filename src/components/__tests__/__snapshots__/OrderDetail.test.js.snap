// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail component matches the snapshot 1`] = `
Array [
  <div>
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div>
        <header
          className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-15 MuiAppBar-colorPrimary mui-fixed"
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <button
              aria-label="Open drawer"
              className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-16 MuiIconButton-edgeStart"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              />
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <div
              className="PrivateHiddenCss-smDown-81"
            >
              <div
                className="MuiBox-root MuiBox-root-92 title"
              >
                <span>
                  Rookie Training 2019
                </span>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-93"
            >
              <div
                className="MuiGrid-root MuiGrid-container"
              >
                <div
                  className="MuiBox-root MuiBox-root-197 makeStyles-navMenuList-20"
                />
                <button
                  aria-controls="long-menu"
                  aria-haspopup="true"
                  aria-label="More"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </header>
      </div>
      <div
        className="makeStyles-toolbar-6"
      />
      <div
        className="MuiBox-root MuiBox-root-222"
      >
        <div
          className="MuiBox-root MuiBox-root-223"
        />
      </div>
      <div
        className="MuiBox-root MuiBox-root-224 txt3"
      >
         
        <span>
          Order History
        </span>
      </div>
      <div>
        <div>
          <div
            className="title_Ship"
          >
            <div
              className="MuiGrid-root nameTxt"
              mb={1}
              mt={1}
            >
              <div
                className="MuiGrid-root txt4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
                ml={1}
              >
                Order Id- 
                 
              </div>
              <div
                className="MuiGrid-root text2 MuiGrid-container"
                mb={1}
                mt={1}
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
                >
                  Order Date- 
                  2019-07-30 07:48:47
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
                >
                  Order Total- 
                  4100
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container"
            mb={1}
            mt={1}
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
            >
              <div
                className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
              >
                <div
                  className="MuiCardContent-root"
                >
                  <div
                    className="MuiBox-root MuiBox-root-227 text2"
                  >
                    <div
                      className="MuiGrid-root txt4 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                      fontSize={20}
                    >
                      Billing Address
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      khin
                       
                      lawoon
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      23street
                       
                      CMTZ
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      mandalay
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      mandalay
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      myanmar
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    />
                    <hr
                      className="MuiDivider-root"
                    />
                    <div
                      className="MuiGrid-root txt4 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      Payment Method
                    </div>
                    <div
                      className="MuiGrid-root text2 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      Checkmoney/MoneyOrder
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
            >
              <div
                className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
              >
                <div
                  className="MuiCardContent-root"
                >
                  <div
                    className="MuiBox-root MuiBox-root-233 text2"
                  >
                    <div
                      className="MuiGrid-root txt4 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                      fontSize={20}
                    >
                      Delivery Address
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      khin
                       
                      lawoon
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      23street
                       
                      CMTZ
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      mandalay
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      mandalay
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      myanmar
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    />
                    <hr
                      className="MuiDivider-root"
                    />
                    <div
                      className="MuiGrid-root txt4 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      Shipping Method
                    </div>
                    <div
                      className="MuiGrid-root text2 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6"
                    >
                      Shipping
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
      >
        <table
          className="MuiTable-root makeStyles-table-2"
        >
          <thead
            className="MuiTableHead-root makeStyles-tbHead-3"
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Id
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                OrderId
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Item Id
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Item Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Price
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Qty
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root makeStyles-cell-4 MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                Total Price
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
          >
            <tr
              className="MuiTableRow-root tbr"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  1
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                />
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  12
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  item1
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  800
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  2
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  1600
                </div>
              </td>
            </tr>
            <tr
              className="MuiTableRow-root tbr"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  2
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                />
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  10
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  item2
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  500
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  5
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  className="text2"
                >
                  2500
                </div>
              </td>
            </tr>
          </tbody>
        </table>
        <div>
          <div
            className="MuiBox-root MuiBox-root-253"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
            >
              <div
                className="MuiBox-root MuiBox-root-254"
              >
                 
              </div>
            </div>
            <div
              className="MuiGrid-root nameTxt MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
            >
              <div
                className="MuiBox-root MuiBox-root-255"
              >
                 
                <span>
                  SubTotal
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-256"
              >
                <span>
                  Sales Tax
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-257"
              >
                <span>
                  Total Price
                </span>
              </div>
            </div>
            <div
              className="MuiGrid-root nameTxt MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
            >
              <div
                className="MuiBox-root MuiBox-root-258"
              >
                4100
                 MMK 
              </div>
              <div
                className="MuiBox-root MuiBox-root-259"
              >
                0.0 MMK 
              </div>
              <div
                className="MuiBox-root MuiBox-root-260"
              >
                4100
                 MMK
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-261"
      >
        <div
          className="MuiBox-root MuiBox-root-262"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span>
                Back
              </span>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;

exports[`OrderDetail component matches the snapshot 2`] = `
Array [
  <div>
    <div>
      <div
        className="MuiBox-root MuiBox-root-298"
      >
        <main
          className="MuiContainer-root container MuiContainer-maxWidthXs"
        >
          <div
            className="MuiBox-root MuiBox-root-301 authTitle"
          >
            <span>
              Sign In
            </span>
          </div>
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <span
              className="close-btn"
            >
              <img
                alt="closeicon"
                src="https://cdn4.iconfinder.com/data/icons/miu/22/circle_close_delete_-128.png"
              />
            </span>
          </a>
          <form
            onSubmit={[Function]}
          >
            <div
              className="MuiBox-root MuiBox-root-338"
            />
            <div
              className="MuiBox-root MuiBox-root-339"
            >
              <div
                className="MuiGrid-root MuiGrid-container"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
                >
                  <div
                    className="MuiBox-root MuiBox-root-444"
                  >
                    <div
                      className="MuiFormControl-root MuiTextField-root txtField MuiFormControl-fullWidth"
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                        data-shrink={false}
                        htmlFor="email"
                        style={
                          Object {
                            "color": "#ffffff",
                          }
                        }
                      >
                        Email Address
                        <span
                          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          autoComplete="email"
                          className="MuiInputBase-input MuiOutlinedInput-input"
                          disabled={false}
                          id="email"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={true}
                          type="email"
                          value=""
                        />
                        <fieldset
                          aria-hidden={true}
                          className="PrivateNotchedOutline-root-502 MuiOutlinedInput-notchedOutline"
                          style={
                            Object {
                              "paddingLeft": 8,
                            }
                          }
                        >
                          <legend
                            className="PrivateNotchedOutline-legend-503"
                            style={
                              Object {
                                "width": 0.01,
                              }
                            }
                          >
                            <span
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "&#8203;",
                                }
                              }
                            />
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
                >
                  <div
                    className="MuiBox-root MuiBox-root-504"
                  >
                    <div
                      className="MuiFormControl-root MuiTextField-root txtField2 MuiFormControl-fullWidth"
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                        data-shrink={false}
                        htmlFor="password"
                        style={
                          Object {
                            "color": "#ffffff",
                          }
                        }
                      >
                        Password
                        <span
                          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          autoComplete="current-password"
                          className="MuiInputBase-input MuiOutlinedInput-input"
                          disabled={false}
                          id="password"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          placeholder="Password"
                          required={true}
                          type="password"
                          value=""
                        />
                        <fieldset
                          aria-hidden={true}
                          className="PrivateNotchedOutline-root-502 MuiOutlinedInput-notchedOutline"
                          style={
                            Object {
                              "paddingLeft": 8,
                            }
                          }
                        >
                          <legend
                            className="PrivateNotchedOutline-legend-503"
                            style={
                              Object {
                                "width": 0.01,
                              }
                            }
                          >
                            <span
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "&#8203;",
                                }
                              }
                            />
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-btnWrapper-299 MuiGrid-container"
            >
              <button
                className="MuiButtonBase-root MuiButton-root btnSign MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="submit"
              >
                <span
                  className="MuiButton-label"
                >
                  <span>
                    Sign In
                  </span>
                </span>
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  className="MuiBox-root MuiBox-root-505"
                >
                  <a
                    className="MuiTypography-root MuiLink-root MuiLink-underlineHover ho MuiTypography-body2 MuiTypography-colorPrimary"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                  >
                    <span>
                      Forgot Password
                    </span>
                  </a>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  className="MuiBox-root MuiBox-root-506"
                >
                  <a
                    className="MuiTypography-root MuiLink-root MuiLink-underlineHover ho MuiTypography-body2 MuiTypography-colorPrimary"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                  >
                    <span>
                      No Account?SignUp?
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </form>
        </main>
      </div>
    </div>
  </div>,
  " ",
]
`;
